cpp/src/slice2cpp/Gen.cpp

```diff
@@ -898,289 +874,184 @@ Slice::Gen::writeExtraHeaders(IceInternal::Output& out)
 void
 Slice::Gen::validateMetadata(const UnitPtr& u)
 {
```

Comment:
Now, instead of creating it's own visitor, `validateMetadata` just creates a map detailing the metadata we care about,
and let's the core `MetadataValidator` handle the rest.